app/scripts/filters/moment.coffee

```diff
@@ -40,7 +40,16 @@ angular.module('swarmApp').filter 'duration', (options, $filter) ->
 # could just pass the template from the view, but this is testable
 angular.module('swarmApp').filter 'warpDuration', ($filter) ->
   (input, unitOfTime, precision) ->
-    $filter('duration')(input, unitOfTime, 'd [days] h [hours and] m [minutes]', precision)
+    if unitOfTime isnt 'seconds'
+      # this function is called only from one place, and right now unitOfTime is always "seconds", but let's fallback elegantly just in case
+      $filter('duration')(input, unitOfTime, 'd [days] h [hours and] m [minutes]', precision)
+    else if Math.abs(input - 15*60) < 1 or input > 60*60
```

Comment:
`input` is usually a decimal.js instance: https://mikemcl.github.io/decimal.js/

decimal.js is used to allow numbers larger than 1e300ish, but durations never get anywhere close to that - so it's safe for lines 15-16 to convert to a Javascript number, and it'd be safe here too. Trying to do math on this without converting first leads to trouble, as you noticed. `toNumber()` first if necessary, and you shouldn't need the stuff with `abs`.

Also, the condition here should probably be part of the `if` above, since they do the same thing.